### ISSUE TITLE
CI?

### ISSUE THREAD

**Issue by till** [byjg/docker-easy-haproxy#3]
I refactored the code and added some tests. Would you enable CI for this repo? I don't know if you have any preference. :)

**Comment by byjg** [comment]
Currently I am using the feature Automated build from Docker Hub to build the images.

But for some Opensource projects I prefer to use the Travis-CI.

**Comment by byjg** [comment]
So, I can enable the Travis-CI.

**Comment by till** [comment]
@byjg Do you have anything against CircleCI? Dealing with Docker is easier. :D I'm almost done. I am building `mode tcp` into your image.

**Comment by byjg** [comment]
Nothing against CircleCI. However I have more than 30 automation using Travis. And Travis is free, unlimited and supports 5 concurrent jobs for opensource projects. CircleCI is also free, but limited to 1 job and 2500 credits (and a "credit" is not exactly "minutes"). 

If you want we can work together. You just open a PR, tell me how are you running the tests and I can configure the CI part. What do you think?

**Comment by till** [comment]
@byjg sounds like a plan!

Right now, I do the following:

```
❯ pytest -s tests/
===================================================================================== test session starts =====================================================================================
platform darwin -- Python 3.7.6, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/local/opt/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /Users/till/Documents/workspaces/pngmbh/docker-easy-haproxy, inifile: pytest.ini
plugins: testinfra-3.4.0
collected 7 items                                                                                                                                                                             

tests/test_labels.py::test_label_generation PASSED
tests/test_labels.py::test_label_data PASSED
tests/test_labels.py::test_label_complex_key PASSED
tests/test_parser.py::test_parser_doesnt_crash PASSED
tests/test_parser.py::test_parser_finds_services PASSED
tests/test_parser.py::test_parser_static PASSED
tests/test_parser.py::test_parser_tcp PASSED

====================================================================================== 7 passed in 0.25s ======================================================================================
```

So it would be nice to have Python 3.7.x and pytest 5.3.x installed? If you get that going, I'll rebase on it. :)

**Comment by till** [comment]
I opened a PR #4 for you to see.

**Comment by byjg** [comment]
I created the branch `2.2` and setup the travis to build this project. 

Once you open a PR pointing to this branch a Job on .travis will start. 

Here is the `.travis.yml` file:
https://github.com/byjg/docker-easy-haproxy/blob/2.2/.travis.yml

Travis Job:
https://travis-ci.org/github/byjg/docker-easy-haproxy

**Comment by byjg** [comment]
Oh, I saw you opened a PR right now. Let me check the build.

**Comment by byjg** [comment]
You have to push something to your branch. I created the Jobs build after you opened the PR :(

**Comment by byjg** [comment]
CI implemented and PR #4 merged. I'll close this issue. 

Thanks :)
